### PR TITLE
fix(umap): handle cuDF strings + improve error messages (#765, #770)

### PR DESCRIPTION
**Pull request by lmeyerov**
## Summary

Fixes two UMAP issues:
- **#765**: UMAP with `engine='cuml'` now works with cuDF DataFrames containing string-typed node ID columns
- **#770**: Better error message when DataFrame has no numeric features after featurization

## Problem 1: cuDF String Node IDs (#765)

When using UMAP with cuDF DataFrames that have string node IDs, the code failed with:
```
TypeError: String arrays are not supported by cupy
```

This happened at `graphistry/umap_utils.py:827` when calling `.values` on a cuDF Series with string dtype (dtype='object'). cuDF tries to convert to a cupy array, but cupy doesn't support string arrays.

## Problem 2: Cryptic Error for All-Strings DataFrames (#770)

When a DataFrame has only string columns (no numeric features), featurization drops all columns and UMAP fails with:
```
ValueError: at least one array or dtype is required
```

This is unhelpful for users who don't know they need numeric columns or skrub installed.

### Fix 1: cuDF String Handling (graphistry/umap_utils.py:827-836)

Added logic to detect cuDF Series with string dtype and convert to pandas before extracting values:

```python
node_series = res._nodes[res._node]
if 'cudf' in str(getmodule(node_series)):
    if node_series.dtype == 'object' or str(node_series.dtype) == 'object':
        logger.debug('Converting cuDF string column to pandas for node extraction')
        nodes = node_series.to_pandas().values
    else:
        import cupy as cp
        nodes = cp.asnumpy(node_series.values)
else:
    nodes = node_series.values
```

### Fix 2: Better Error Message (graphistry/umap_utils.py:851-858, 891-898)

Added validation after featurization to provide helpful guidance:

```python
if len(X_.columns) == 0:
    raise ValueError(
        "No numeric features available for UMAP after featurization. "
        "All non-numeric columns were dropped. "
        "Please provide at least one numeric column, or install 'skrub' for automatic string encoding: "
        "pip install skrub"
    )
```

## Testing

**✅ All 11/11 comprehensive tests passed** (pandas + cuDF, CPU + GPU)

### Test Matrix

**Part 1: String IDs + Numeric Features**
- ✅ Test 1.1 (pandas + umap_learn): SUCCESS
- ✅ Test 1.2 (pandas + cuml): SUCCESS  
- ✅ Test 1.3 (cudf + cuml): SUCCESS

**Part 2: Numeric IDs + String Features**
- ✅ Test 2.1 (pandas + umap_learn): SUCCESS
- ✅ Test 2.2 (pandas + cuml): SUCCESS
- ✅ Test 2.3 (cudf + cuml): SUCCESS

**Part 3: String IDs + String Features (ORIGINAL BUG #765)**
- ✅ Test 3.1 (pandas + umap_learn): SUCCESS
- ✅ Test 3.2 (pandas + cuml): SUCCESS
- ✅ **Test 3.3 (cudf + cuml) - ORIGINAL BUG**: SUCCESS ← **THIS WAS THE FAILING CASE!**

**Part 4: All-Strings Edge Case (Issue #770)**
- ✅ Test 4.1 (pandas + umap_learn): SUCCESS
- ✅ Test 4.2 (cudf + cuml): SUCCESS ← **Issue #770 verified fixed!**

### Test Environment
- Local: Python 3.8, pandas (3/11 tests)
- Docker GPU: Python 3.10, cuDF, cuML, RAPIDS (11/11 tests)

## Impact

Enables very common use cases:
1. UMAP with `engine='cuml'` on cuDF DataFrames with string node IDs
2. Provides clear guidance when users forget to include numeric columns

## Files Modified

- `graphistry/umap_utils.py`: Node ID extraction fix + validation
- `CHANGELOG.md`: Documented both fixes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: graphistry/umap_utils.py [hunk 1]
```diff
@@ -876,7 +888,16 @@ def umap(
...
             )
 
-            # add the safe coercion here 
+            # Validate that we have at least one feature column for UMAP
+            if len(X_.columns) == 0:
+                raise ValueError(
+                    "No numeric features available for UMAP after featurization. "
+                    "All non-numeric columns were dropped. "
+                    "Please provide at least one numeric column, or install 'skrub' for automatic string encoding: "
+                    "pip install skrub"
```
Comment (lmeyerov) [inline graphistry/umap_utils.py]: bad msg
